src/startup.ts

```diff
@@ -1,60 +1,89 @@
-		action,
-		rtorrentRpcUrl,
-		qbittorrentUrl,
-		transmissionRpcUrl,
-		delugeRpcUrl,
-		dataDirs,
-		linkDir,
-		matchMode,
-		skipRecheck,
-	} = getRuntimeConfig();
-	if (
-		action === "inject" &&
-		!(
-			rtorrentRpcUrl ||
-			qbittorrentUrl ||
-			transmissionRpcUrl ||
-			delugeRpcUrl
-		)
-	) {
+	const downloadClient = getClient();
+	await Promise.all<void>([
+		validateTorznabUrls(),
+		downloadClient?.validateConfig(),
+	]);
+	logger.verbose({
+		label: Label.CONFIGDUMP,
+		message: inspect(runtimeConfig),
+	});
+	if (!runtimeConfig.configValid) {
 		throw new CrossSeedError(
-			"You need to specify --rtorrent-rpc-url, --transmission-rpc-url, --qbittorrent-url, or --deluge-rpc-url when using '-A inject'."
+			`Your configuration is invalid, please see the ${
+				zodErrorCount > 1 || (zodErrorCount > 0 && !validConfigPaths)
+					? "errors"
+					: "error"
+			} above for details.`
 		);
 	}
-	if ((dataDirs && !linkDir) || (!dataDirs && linkDir)) {
-		throw new CrossSeedError(
-			"Data based matching requires both --link-dir and --data-dirs"
-		);
+	logger.info("Your configuration is valid!");
+}
+/**
+ * verifies the config paths provided against the filesystem
+ * @returns true (if paths are valid)
+ */
+function checkConfigPaths(): boolean {
+	const { action, linkDir, dataDirs, torrentDir, outputDir } =
+		getRuntimeConfig();
+
+	let pathFailure = false;
+	if (!existsSync(torrentDir)) {
+		reportBadPath("torrentDir", torrentDir);
+		pathFailure = true;
 	}
-	if (matchMode == MatchMode.RISKY && skipRecheck) {
-		logger.warn(
-			"It is strongly recommended to not skip rechecking for risky matching mode"
-		);
+
+	if (action == Action.SAVE && !existsSync(outputDir)) {
+		reportBadPath("outputDir", outputDir);
+		pathFailure = true;
 	}
-}
 
-export async function doStartupValidation(): Promise<void> {
-	logger.info("Validating your configuration...");
-	validateOptions();
-	const downloadClient = getClient();
-	await Promise.all<void>(
-		[
-			validateAction(),
-			validateTorznabUrls(),
-			downloadClient?.validateConfig(),
-			validateTorrentDir(),
-		].filter(Boolean)
-	);
-	logger.info("Your configuration is valid!");
+	if (linkDir && !existsSync(linkDir)) {
+		reportBadPath("linkDir", linkDir);
+		pathFailure = true;
+	}
+	if (dataDirs) {
+		for (const dataDir of dataDirs) {
+			if (!existsSync(dataDir)) {
+				reportBadPath("dataDirs", dataDir);
+				pathFailure = true;
+			}
+		}
+	}
+	return !pathFailure;
+}
+/**
+ * logs an error message for invalid path settings (either filesystem or formatting)
+ * @param configSetting (the name of the option set)
+ * @param configValue (the value the option is set to)
+ */
+function reportBadPath(configSetting: string, configValue: string) {
+	const pathError =
+		sep === "\\" &&
+		!configValue.includes("\\") &&
+		!configValue.includes("/")
+			? `Your ${configSetting} "${configValue}" is not formatted properly for Windows. Please use "\\\\" or "/" for directory separators.\n`
```

Comment:
this is a creative check to do! seems like we don't actually need fs access for it though - I think we could use a `refine` in the schema similar to what we do with the `ms` values to handle this case